pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -861,6 +876,23 @@ func loadBalancerServiceIsProgressing(ic *operatorv1.IngressController, service
 		}
 	}
 
+	if platform.Type == configv1.OpenStackPlatformType {
+		var (
+			wantFloatingIP, haveFloatingIP string
+		)
+		if fip := getOpenStackFloatingIPInSpec(ic); fip != "" {
+			wantFloatingIP = fip
+		}
+		if fip := getOpenStackFloatingIPInStatus(ic); fip != "" {
+			haveFloatingIP = fip
+		}
+		// OpenStack CCM does not support updating Service.Spec.LoadBalancerIP after creation, the load balancer will never be updated.
+		if wantFloatingIP != haveFloatingIP {
+			err := fmt.Errorf("The IngressController floatingIP was changed from %q to %q.", haveFloatingIP, wantFloatingIP)
```

Comment:
I'll add the instructions in this PR and a TODO to use your new function once it has merged.

Reply:
That's fine, but I expect to land that PR in the next couple weeks, but we aren't gated by FF date for that bug. If you'd like this PR by FF, then you might want to avoid the dependency on that PR.

Reply:
Would it be possible to address that comment after FF, since your PR is a bugfix, so I can address that via a bug post FF as well? I want to avoid copying your unmerged code and create complexity. I'll come back to this for sure once your PR has merged.

Reply:
I'd prefer to add the message in this PR if possible, as we've considered missing condition messages a bug in the past.

No worries on pulling in my code, instead follow the existing patterns for subnet and EIPs, and I'll clean it up later when I introduce the function. Shouldn't be too bad to implement, but let me know if you have issues.

Reply:
good, I've added the message following subnet and EIPs's patterns.